com.unity.render-pipelines.universal/Runtime/Passes/DrawSkyboxPass.cs

```diff
@@ -15,7 +15,48 @@ public DrawSkyboxPass(RenderPassEvent evt)
         /// <inheritdoc/>
         public override void Execute(ScriptableRenderContext context, ref RenderingData renderingData)
         {
-            context.DrawSkybox(renderingData.cameraData.camera);
+#if ENABLE_VR && ENABLE_XR_MODULE
+            // XRTODO: Remove this code once Skybox pass is moved to SRP land.
+            if (renderingData.cameraData.xr.enabled)
+            {
+                // Setup Legacy XR buffer states
+                if (renderingData.cameraData.xr.singlePassEnabled)
+                {
+                    // Setup legacy skybox stereo buffer
+                    renderingData.cameraData.camera.SetStereoProjectionMatrix(Camera.StereoscopicEye.Left, renderingData.cameraData.xr.GetProjMatrix(0));
+                    renderingData.cameraData.camera.SetStereoViewMatrix(Camera.StereoscopicEye.Left, renderingData.cameraData.xr.GetViewMatrix(0));
+                    renderingData.cameraData.camera.SetStereoProjectionMatrix(Camera.StereoscopicEye.Right, renderingData.cameraData.xr.GetProjMatrix(1));
+                    renderingData.cameraData.camera.SetStereoViewMatrix(Camera.StereoscopicEye.Right, renderingData.cameraData.xr.GetViewMatrix(1));
+
+                    CommandBuffer cmd = CommandBufferPool.Get();
+
+                    // Use legacy stereo instancing mode to have legacy XR code path configured
+                    cmd.SetSinglePassStereo(Application.platform == RuntimePlatform.Android ? SinglePassStereoMode.Multiview : SinglePassStereoMode.Instancing);
+                    context.ExecuteCommandBuffer(cmd);
+                    cmd.Clear();
+
+                    // Calling into built-in skybox pass
+                    context.DrawSkybox(renderingData.cameraData.camera);
+
```

Comment:
1. Mixing old legacy XR code makes this "impure" XR. What's the schedule for porting skybox to pure SRP?
2. Calling cmd.SetSinglePassStereo with either multiview or SPI assumes there's no multpass mode. This would break in multipass mode.